syntax = "proto3";

package signal;


message Complex {
    double real = 1;
    double imag = 2;
}


message FirstDifferenceRequest {
    repeated Complex signal = 1;      
}


message FirstDifferenceResponse {
    repeated Complex result = 1; 
}


service SignalService {
    rpc ComputeFirstDifference (FirstDifferenceRequest) returns (FirstDifferenceResponse);
}